pyroute2/ipdb/routes.py

```diff
@@ -134,14 +134,19 @@ def __init__(self, route):
 
 
 # Universal route key
+# Holds the fields that the kernel uses to uniquely identify routes.
+# IPv4 allows redundant routes with different 'tos' but IPv6 does not,
+# so 'tos' is used for IPv4 but not IPv6.
+# For reference, see fib_table_insert() in
+# https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/ipv4/fib_trie.c#n1147
+# and fib6_add_rt2node() in
+# https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/ipv6/ip6_fib.c#n765
 RouteKey = namedtuple('RouteKey',
                       ('dst',
-                       'scope',
```

Comment:
so the scope is not used as a key?

… I'll try to find a relevant issue, but I believe it was added here after a bug record.

Reply:
Correct, scope is not used as a key.  You can verify by looking at the kernel source (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/net/ipv4/fib_trie.c#n1147), or using iproute2:
# ip route add 123.0.0.0/24 via 192.168.0.1 scope 10
# ip route add 123.0.0.0/24 via 192.168.0.1 priority 100
# ip route add 123.0.0.0/24 via 192.168.0.1 tos 1
# ip route
123.0.0.0/24 tos 0x01 via 192.168.0.1 dev eth0 
123.0.0.0/24 via 192.168.0.1 dev eth0 scope 10
123.0.0.0/24 via 192.168.0.1 dev eth0 metric 100
# ip route add 123.0.0.0/24 via 192.168.0.1 scope 11
RTNETLINK answers: File exists
# ip route del 123.0.0.0/24 via 192.168.0.1 scope 10
# ip route add 123.0.0.0/24 via 192.168.0.1 scope 11
# ip route
123.0.0.0/24 tos 0x01 via 192.168.0.1 dev eth0 
123.0.0.0/24 via 192.168.0.1 dev eth0 scope 11
123.0.0.0/24 via 192.168.0.1 dev eth0 metric 100

Reply:
Okay. Thanks!

Reply:
It looks like the bug you are referring to is #310, and the relevant commit is https://github.com/svinota/pyroute2/commit/ae1eb7e80e7ada65ea98b4b50845246c01883c33.

It looks to me like the important part of this commit (the part which actually fixed the issue with changing priority on non-default routes) was:
```
+        if 'scope' not in spec:
+            spec['scope'] = 0
```

Adding 'scope' to RouteKey was not actually required to fix that issue.